### PR TITLE
Add Config Merger for k8s in Canary

### PR DESCRIPTION
**Pull request by chases2**
Also allows config_merger to read from a URL with a special flag.

Do not merge until https://github.com/kubernetes/test-infra/pull/20986 merges, or the URL won't work.

### CODE REVIEW

File: cluster/canary/config_merger.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        app: testgrid
+        channel: stable
+        component: config-merger
+    spec:
+      serviceAccountName: config-merger
+      containers:
+      - name: config-merger
+        image: gcr.io/k8s-testgrid/config_merger:v20210309-v0.0.54-12-g4289ea4
+        args:
+        - --config-url="https://raw.githubusercontent.com/kubernetes/test-infra/master/testgrid/mergelists/canary.yaml"
```
Comment (chases2) [inline cluster/canary/config_merger.yaml:30]: Note: I moved this, so the URL will be different. Will fix after other PR merges.

**Review by michelle192837 — APPROVED** [review]
/hold

Unhold at will!

**Review by e-blackwelder — APPROVED** [review]

**Pull request merged**